File: api/Woodshed.Application/Models/Response/Identity/UserResponse.cs
namespace Woodshed.Application.Models.Response.Identity;

public class UserResponse
{
    public string Id { get; set; } = string.Empty;
    public string NickName { get; set; } = string.Empty;
    public string? Email { get; set; } = string.Empty;
    public string? Name { get; set; } = string.Empty;
    public string? LastName { get; set; } = string.Empty;
    public string? Biography { get; set; } = string.Empty;
    public string? ImageUrl { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
}

File: client/src/shared/schemas/response/UserResponse.ts
import { z } from "zod"

export const UserResponseSchema = z.object({
  id: z.string(),
  nickName: z.string(),
  email: z.string(),
  name: z.string(),
  lastName: z.string(),
  biography: z.string(),
  imageUrl: z.string(),
  createdAt: z.iso.datetime(),
})

export type UserResponse = z.infer<typeof UserResponseSchema>